### PR TITLE
MRG: Deprecate silently adding a leading period to an extension in BIDSPath

### PR DESCRIPTION
**Pull request by hoechenberger**
This works in `main`:
```python
# %%
from mne_bids import BIDSPath

bp_1 = BIDSPath(extension='vhdr')
bp_2 = BIDSPath(extension='.vhdr')

assert bp_1 == bp_2
```

With this PR, it warns:

> FutureWarning: extension should start with a period ".", but got: "vhdr". Prepending "." to form: ".vhdr". This will raise an exception starting with MNE-BIDS 0.12.

This is part of my journey to try and make MNE-BIDS less clever.

Merge checklist
---------------

Maintainer, please confirm the following before merging.
If applicable:

- [x] All comments are resolved
- [x] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [x] New contributors have been added to [CITATION.cff](https://github.com/mne-tools/mne-bids/blob/main/CITATION.cff)
- [x] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: doc/whats_new.rst [hunk 1]
```diff
@@ -72,6 +72,8 @@ Detailed list of changes
 
 - Passing only one of ``events`` and ``event_id`` to :func:`~mne_bids.write_raw_bids` now raises a ``ValueError`` instead of a ``RuntimeError``, by `Richard Höchenberger`_ (:gh:`1054`)
 
+- Until now, :class:`mne_bids.BIDSPath` prepends extensions with a period "." automatically. We intend to remove this undocumented side-effect and now emit a ``FutureWarning`` if an ``extension`` that does not start with a ``.`` is provided. Starting with MNE-BIDS 0.12, an exeption will be raised in this case, by `Richard Höchenberger`_ (:gh:`xxx`)
```
Comment (hoechenberger) [inline doc/whats_new.rst:75]:
```suggestion
- Until now, :class:`mne_bids.BIDSPath` prepends extensions with a period "." automatically. We intend to remove this undocumented side-effect and now emit a ``FutureWarning`` if an ``extension`` that does not start with a ``.`` is provided. Starting with MNE-BIDS 0.12, an exception will be raised in this case, by `Richard Höchenberger`_ (:gh:`xxx`)
```

**Review by adam2392 — APPROVED** [review]

**Comment by adam2392** [comment]
Yay for less magic :)

**Review by agramfort — APPROVED** [review]

**Comment by sappelhoff** [comment]
it's funny how the windows job with the NIRS validator passes :exploding_head:  ... ubuntu and macos don't

now I am re-running after a potential fix has been pushed upstream

**Review by sappelhoff — APPROVED** [review]
thanks! Makes sense to me. Also BIDS says:

>  Note that the left-most period is included in the file extension.

https://bids-specification.readthedocs.io/en/latest/02-common-principles.html

**Pull request merged**